src/pathpyG/core/TemporalGraph.py

```diff
@@ -111,82 +90,81 @@ def from_csv(file, timestamp_format='%Y-%m-%d %H:%M:%S', time_rescale=1) -> Temp
+        """Randomly shuffles the temporal order of edges by randomly permuting timestamps."""
         self.data['t'] = self.data['t'][torch.randperm(len(self.data['t']))]
         # t_sorted, indices = torch.sort(torch.tensor(t).to(config["torch"]["device"]))
         # self.data['src'] = self.data['src']
         # self.data['dst'] = self.data['dst']
         # self.data['t'] = t_sorted
 
-    def to_static_graph(self) -> Graph:
-        """Return instance of [`Graph`][pathpyG.Graph] that represents the static, time-aggregated network.
-        """
-        node_id = [self.node_index_to_id[i] for i in range(self.N)]
-        return Graph(self.data.edge_index, node_id)
-
-    def to_pyg_data(self) -> TemporalData:
-        """
-        Returns an instance of [`torch_geometric.Data`](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.data.Data.html#torch_geometric.data.Data) containing the
-        `edge_index` as well as node, edge, and graph attributes
+    def to_static_graph(self, weighted=False, time_window: Optional[Tuple[int,int]]=None) -> Graph:
+        """Return weighted time-aggregated instance of [`Graph`][pathpyG.Graph] graph.
         """
-        return self.data
-
+        if time_window is not None:
+            index = (self.data.t >= time_window[0]).logical_and(self.data.t < time_window[1]).nonzero().ravel()
+            edge_index = torch.stack((self.data.src[index], self.data.dst[index]))
+        else:
+            edge_index = self.data.edge_index
+        if weighted:
+            i, w = torch_geometric.utils.coalesce(edge_index, torch.ones(self.M))
+            return Graph(Data(edge_index=i, edge_weight=w), self.mapping)
+        else:
+            return Graph.from_edge_index(edge_index, self.mapping)
 
-    def get_window(self, start, end):
+    def get_window(self, start: int, end: int) -> TemporalGraph:
         """Returns an instance of the TemporalGraph that captures all time-stamped 
         edges in a given window defined by start and (non-inclusive) end, where start
         and end refer to the number of events"""
 
-        idx = torch.tensor([self.data['src'][start:end].numpy(), self.data['dst'][start:end].numpy()]).to(config["torch"]["device"])
-        max_idx = torch.max(idx).item()
+        #idx = torch.tensor([self.data['src'][start:end].numpy(), self.data['dst'][start:end].numpy()]).to(config["torch"]["device"])
+        #max_idx = torch.max(idx).item()
+
+        # return TemporalGraph(
+        #     edge_index = idx,
+        #     t = self.data.t[start:end],
+        #     node_id = self.data.node_id[:max_idx+1]
+        #     )
+        return TemporalGraph(
+            data = TemporalData(
+                src=self.data.src[start:end],
+                dst=self.data.dst[start:end],
+                t=self.data.t[start:end]
+            ),
+            mapping = self.mapping
+        )
+    
+
+    def get_snapshot(self, start: int, end: int) -> TemporalGraph:
```

Comment:
Is this an alias of `get_window()`?

Reply:
Actually no! `get_window` returns a `TemporalGraph` object that only contains time-stamped edges in a given time window. `get_snapshot` returns a static (potentially) weighted time-aggregated graph, i.e. an instance of class `Graph`